tests/unit/ast/test_pre_parser.py

```diff
@@ -224,3 +225,37 @@ def test_parse_pragmas(code, pre_parse_settings, compiler_data_settings, mock_ve
 def test_invalid_pragma(code):
     with pytest.raises(StructureException):
         pre_parse(code)
+
+
+def test_version_conflict_with_imports(make_input_bundle, mock_version):
+    lib_version = "~=0.3.10"
+    contract_version = "0.4.0"
```

Comment:
i'm not sure this is necessary (it's also brittle, the test will start failing for the wrong reason after 0.4.0 release)